### PR TITLE
update OptionalStringListValue and refactor code using it

### PR DESCRIPTION
**Pull request by davidbloss**
## Changelog

Simplify `OptionalStringListValue()` and as a result much of the code calling it.

- [X] List your changes here
- [ ] Make a `changie` entry, N/A internal changes only

## Tophatting

[task test-integration](https://github.com/OpsLevel/terraform-provider-opslevel/actions/runs/9911184718/job/27383271780)

### CODE REVIEW

File: .changes/unreleased/Refactor-20240710-105829.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+kind: Refactor
+body: improve OptionalStringListValue(), dropped context input arg and diagnostics ouput arg
```
Comment (rocktavious) [inline .changes/unreleased/Refactor-20240710-105829.yaml:2]: This seems like a fully internal change and likely won't require a customer facing change entry

File: opslevel/terraform_type_conversions.go [hunk 2]
```diff
@@ -65,11 +65,17 @@ func RequiredIntValue(value int) basetypes.Int64Value {
...
+		return types.ListNull(types.StringType)
 	}
-	return types.ListValueFrom(ctx, types.StringType, value)
+
+	elems := make([]attr.Value, len(values))
+	for i, v := range values {
+		elems[i] = types.StringValue(v)
+	}
+
+	return types.ListValueMust(types.StringType, elems)
```
Comment (rocktavious) [inline opslevel/terraform_type_conversions.go:78]: Does this just `panic` then?

File: .changes/unreleased/Refactor-20240710-105829.yaml (hunk 1 above)
Comment (davidbloss) [reply .changes/unreleased/Refactor-20240710-105829.yaml:2]: True, I can delete this

File: opslevel/terraform_type_conversions.go (hunk 2 above)
Comment (davidbloss) [reply opslevel/terraform_type_conversions.go:78]: There is a panic but not one we would worry about 😁 

[Under the hood](https://github.com/hashicorp/terraform-plugin-framework/blob/main/types/basetypes/list_value.go#L72-L83) the panic would be thrown if the type of an element in `elems` does not match `types.StringType`. But this is covered at line 75

**Review by rocktavious — APPROVED** [review]

**Pull request merged**